core/amber/src/main/scala/edu/uci/ics/amber/engine/common/promise/RPCClient.scala

```diff
@@ -0,0 +1,53 @@
+package edu.uci.ics.amber.engine.common.promise
+
+import com.twitter.util.{Future, Promise}
+import edu.uci.ics.amber.engine.architecture.messaginglayer.ControlOutputPort
+import edu.uci.ics.amber.engine.common.ambermessage.neo.ControlPayload
+import edu.uci.ics.amber.engine.common.ambertag.neo.VirtualIdentity.ActorVirtualIdentity
+import edu.uci.ics.amber.engine.common.promise.RPCClient.{ControlInvocation, ReturnPayload}
+import edu.uci.ics.amber.engine.common.promise.RPCServer.RPCCommand
+
+import scala.collection.mutable
+
+object RPCClient {
+
+  /** The invocation of a control command
+    * @param id
+    * @param call
+    */
+  case class ControlInvocation(id: Long, call: RPCCommand[_]) extends ControlPayload
+
+  /** The return message of a promise.
+    * @param returnValue
+    */
+  case class ReturnPayload(id: Long, returnValue: Any) extends ControlPayload
+
+}
+
+class RPCClient(controlOutputPort: ControlOutputPort) {
```

Comment:
the best name I am able to come up with is ControlMessageSource or ControlInvocationSource.

Reply:
Corresponding term for RPCServer is ControlMessageReceiver